aquarium-native/src/InnerModel.h

```diff
@@ -0,0 +1,15 @@
+// InnerModel.h: Define inner model.
+
+#pragma once
+#ifndef INNERMODEL_H
+#define INNERMODEL_H 1
+
+#include "Model.h"
+
+class InnerModel : public Model

```

Comment:
Adding multiple subclasses of Model doesn't match the structure of the JavaScript code well. If the code runs faster this way then maybe the JavaScript code would be faster if it were unrolled in this way. Right now this is a *lot* more code – and has a lot more repetition – than the JavaScript port. Would you please try to match the structure of the JS code better? The goal of this port, after all, is to have more or less a side-by-side comparison of a medium-sized example.

Reply:
Yes, I agree. This one has optimized the structure and we shouldn't compare the version with JS one.Actually, we want to compare the performance gap between native and web, and also between different backends.
Another logically mapping version has similar data structure with the js version. Should I upload that one first?

Reply:
@JiangYizhou , I replied to the thread above.